job/job.go

```diff
@@ -206,16 +173,23 @@ func (j *Job) InitDelayDuration(checkTime bool) error {
 }
 
 // StartWaiting begins a timer for when it should execute the Jobs .Run() method.
-func (j *Job) StartWaiting() {
+func (j *Job) StartWaiting(cache JobCache) {
 	waitDuration := time.Duration(j.scheduleTime.UnixNano() - time.Now().UnixNano())
 	log.Debug("Wait Duration initial: %s", waitDuration)
 	if waitDuration < 0 {
 		// Needs to be recalculated each time because of Months.
-		waitDuration = j.delayDuration.ToDuration()
+		lastRun := j.LastAttemptedRun
+		lastRun.Add(j.delayDuration.ToDuration())
+		waitDuration = time.Duration(lastRun.UnixNano() - time.Now().UnixNano())
 	}
 	log.Info("Job Scheduled to run in: %s", waitDuration)
 	j.NextRunAt = time.Now().Add(waitDuration)
-	j.jobTimer = time.AfterFunc(waitDuration, j.Run)
+	jobRun := func(cache JobCache) func() {
+		return func() {
+			j.Run(cache)
+		}
+	}(cache)
```

Comment:
@ajvb `cache` isn't being defined on every iteration of a for-loop or anything, so this is good enough:

```go
jobRun := func() { j.Run(cache) }
```